src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -149,89 +149,90 @@ internal Metadata GetMetadata(string fullPath, DateTime snapshotTimestamp)
             GC.KeepAlive(newMetadata);
 
             return metadata;
-        }
 
-        /// <exception cref="IOException"/>
-        /// <exception cref="BadImageFormatException" />
-        private AssemblyMetadata CreateAssemblyMetadataFromTemporaryStorage(
-            FileKey fileKey, List<TemporaryStorageService.TemporaryStreamStorage> storages)
-        {
-            var moduleMetadata = CreateModuleMetadataFromTemporaryStorage(fileKey, storages);
-            return CreateAssemblyMetadata(fileKey, moduleMetadata, storages, CreateModuleMetadataFromTemporaryStorage);
+            // <exception cref="IOException"/>
+            // <exception cref="BadImageFormatException" />
+            AssemblyMetadata CreateAssemblyMetadataFromTemporaryStorage()
```

Comment:
>CreateAssemblyMetadataFromTemporaryStorage [](http://example.com/codeflow?start=29&length=42)

I'd remove the local functions and just inline it all the way.

Reply:
k :)